### PR TITLE
feat: Linea mainnet deployment

### PR DESCRIPTION
**Pull request by sanchaymittal**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

**Review by just-a-node — APPROVED** [review]

File: packages/deployments/subgraph/config/amarok-hub-v0.json [hunk 1]
```diff
@@ -37,6 +37,11 @@
         "name": "BnbHubConnector",
         "address": "0x69009c6f567590d8B469dbF4C8808e8ee32b8a45",
         "startBlock": 17640454
+      },
+      {
+        "name": "LineaHubConnector",
+        "address": "0x076cD2B25cb1Ed7272d716bdeb4A8551CF606a3D",
+        "startBlock": 18468809
```
Comment (wanglonghong) [inline packages/deployments/subgraph/config/amarok-hub-v0.json:44]: I don't think we need `LineaHubConnector` address here if we don't have a network entity in subgraph.template.yaml. or maybe we are currently missing it in the template file???

**Review by wanglonghong — COMMENTED** [review]
One thing more we have to take care of after the merge into prod is the gasEstimates values in chaindata repo. because it hasn't set up yet but also understandable right now. but we should definitely set the proper values as soon as we go live.

**Review by preethamr — DISMISSED** [review]

File: ops/mainnet/prod/backend/config.tf [hunk 2]
```diff
@@ -50,6 +50,9 @@ locals {
       "6778479" = {
         providers = ["https://rpc.ankr.com/gnosis"]
       }
+      "1818848877" = {
+        providers = ["https://rpc.linea.build"]
+      }
```
Comment (LayneHaber) [inline ops/mainnet/prod/backend/config.tf:55]: wont we need a more stable rpc for the expected load?

Comment (LayneHaber) [reply ops/mainnet/prod/backend/config.tf:55]: guess its more important for core

File: packages/deployments/contracts/deployConfig/shared.ts [hunk 3]
```diff
@@ -476,6 +476,19 @@ export const MESSAGING_PROTOCOL_CONFIGS: Record<string, MessagingProtocolConfig>
...
+        prefix: "Linea",
+        ambs: {
+          // https://etherscan.io/address/0xd19d4B5d358258f05D7B411E21A1460D11B0876F
+          hub: "0xd19d4B5d358258f05D7B411E21A1460D11B0876F",
+          // https://lineascan.build/address/0x508Ca82Df566dCD1B0DE8296e70a96332cD644ec
+          spoke: "0x508Ca82Df566dCD1B0DE8296e70a96332cD644ec",
+        },
+        processGas: DEFAULT_PROCESS_GAS,
+        reserveGas: DEFAULT_RESERVE_GAS,
+        delayBlocks: DEFAULT_DELAY_BLOCKS,
```
Comment (LayneHaber) [inline packages/deployments/contracts/deployConfig/shared.ts:490]: will these defaults be sufficient?

File: packages/deployments/contracts/src/cli/init/config/mainnet/production.ts [hunk 4]
```diff
@@ -397,6 +414,7 @@ export const MAINNET_PRODUCTION_INIT_CONFIG: InitConfig = {
       "1634886255": "0x5C711DB90dEc0a5B81C626968DEa4187a7f9C1F2",
       "6450786": "0x9435Ba7C661a0Fd477deED640491de8c100325A7",
       "6778479": "0x7616Bc6d0dee5E250BA5b3dDa6cbbB71786FB638",
+      "1818848877": "0xade09131C6f43fe22C2CbABb759636C43cFc181e",
```
Comment (LayneHaber) [inline packages/deployments/contracts/src/cli/init/config/mainnet/production.ts:417]: this should be the protocol admin safe: `0x50d22a34183e7cD3164184011998a3D50303f75d `

File: packages/deployments/contracts/src/cli/init/config/mainnet/production.ts [hunk 5]
```diff
@@ -397,6 +414,7 @@ export const MAINNET_PRODUCTION_INIT_CONFIG: InitConfig = {
...
     },
     watchers: {
       allowlist: ["0xade09131C6f43fe22C2CbABb759636C43cFc181e"],
```
Comment (LayneHaber) [inline packages/deployments/contracts/src/cli/init/config/mainnet/production.ts:420]: we are missing addresses of our mainnet deployed watchers here, need to add them

File: packages/deployments/contracts/deployConfig/shared.ts (hunk 3 above)
Comment (LayneHaber) [reply packages/deployments/contracts/deployConfig/shared.ts:490]: it will require owner to change, so we should make sure they are in advance

**Comment by LayneHaber** [comment]
> One thing more we have to take care of after the merge into prod is the gasEstimates values in chaindata repo. because it hasn't set up yet but also understandable right now. but we should definitely set the proper values as soon as we go live.

we also have to add the assets properly to make sure they are included in the watcher v2 implementation

File: packages/deployments/contracts/deployConfig/shared.ts (hunk 3 above)
Comment (LayneHaber) [reply packages/deployments/contracts/deployConfig/shared.ts:490]: (if this is on the hub, the owner will be the SC once transferred)

File: packages/deployments/contracts/src/cli/init/config/mainnet/production.ts (hunk 5 above)
Comment (preethamr) [reply packages/deployments/contracts/src/cli/init/config/mainnet/production.ts:420]: Updated

File: packages/deployments/contracts/src/cli/init/config/mainnet/production.ts (hunk 4 above)
Comment (preethamr) [reply packages/deployments/contracts/src/cli/init/config/mainnet/production.ts]: Updated

File: packages/deployments/contracts/src/cli/init/config/mainnet/production.ts (hunk 5 above)
Comment (preethamr) [reply packages/deployments/contracts/src/cli/init/config/mainnet/production.ts:420]: > we are missing addresses of our mainnet deployed watchers here, need to add them

What other new watchers do we have, the V2 watchers ? @LayneHaber

File: ops/mainnet/prod/backend/config.tf (hunk 2 above)
Comment (preethamr) [reply ops/mainnet/prod/backend/config.tf:55]: Yes, as SDK server is only using public RPCs.

File: packages/deployments/contracts/src/cli/init/config/mainnet/production.ts (hunk 5 above)
Comment (LayneHaber) [reply packages/deployments/contracts/src/cli/init/config/mainnet/production.ts:420]: added ours, requested info from others. we can add with protocol admin, we should wait to whitelist until they are confirmed updated with the latest version

File: packages/deployments/contracts/deployConfig/shared.ts (hunk 3 above)
Comment (LayneHaber) [reply packages/deployments/contracts/deployConfig/shared.ts:490]: verified against other spoke connectors

File: packages/deployments/subgraph/config/amarok-hub-v0.json (hunk 1 above)
Comment (LayneHaber) [reply packages/deployments/subgraph/config/amarok-hub-v0.json:44]: it is in the template file: https://github.com/connext/monorepo/blob/fad4783676bfa7124092b9d86dc691a89871234b/packages/deployments/subgraph/src/amarok-hub-v0/subgraph.template.yaml#L221-L247

**Comment by LayneHaber** [comment]
chaindata on main already has the watcher flags

I ensured the multi minter branch used by the watcher is up to date as well

@wanglonghong are there other gas estimates that need to be added? I see one for `execute` but im not sure how accurate it is: https://github.com/connext/chaindata/blob/fc0f11f0e25ffd4db20ca9839ef563764f533a02/crossChain.json#L2555

**Comment by LayneHaber** [comment]
> we also have to add the assets properly to make sure they are included in the watcher v2 implementation

done

**Review by LayneHaber — APPROVED** [review]

**Review by preethamr — APPROVED** [review]

**Comment by wanglonghong** [comment]
> chaindata on main already has the watcher flags
> 
> I ensured the multi minter branch used by the watcher is up to date as well
> 
> @wanglonghong are there other gas estimates that need to be added? I see one for `execute` but im not sure how accurate it is: https://github.com/connext/chaindata/blob/fc0f11f0e25ffd4db20ca9839ef563764f533a02/crossChain.json#L2555

What we need is `execute` and `gasPriceFactor` but the default value (1e18) will be used if unset. so that's fine. 
In terms of accuracy, I think we can give it a try with 350000.

**Review by wanglonghong — APPROVED** [review]

**Review by liu-zhipeng — APPROVED** [review]

**Review by just-a-node — APPROVED** [review]

**Pull request merged**